code/modules/vehicles/_vehicle.dm

```diff
@@ -111,7 +112,14 @@
 	if(lastmove + movedelay > world.time)
 		return FALSE
 	lastmove = world.time
-	return step(src, direction)
+	if(trailer)
+		var/dir_to_move = get_dir(trailer.loc, loc)
+		var/did_move = step(src, direction)
+		if(did_move)
+			step(trailer, dir_to_move)
```

Comment:
you should call move on the trailer so it gets proper movement

Reply:
(step calls move)